String/AlphaNumericPalindrome.js

```diff
@@ -1,32 +1,35 @@
-/*****************************************************************************
- * @function alphaNumericPlaindrome
- * @description alphaNumericPlaindrome should return true if the string has alphanumeric characters that are palindrome irrespective of special characters and the letter case.
+/**
+ * @function alphaNumericPalindrome
+ * @description alphaNumericPalindrome should return true if the string has alphanumeric characters that are palindrome irrespective of special characters and the letter case.
  * @param {string} str the string to check
- * @returns {Boolean}
- * @see [Factorial](https://en.wikipedia.org/wiki/Palindrome)
+ * @returns {boolean}
+ * @see [Palindrome](https://en.wikipedia.org/wiki/Palindrome)
  * @example
- * The function alphaNumericPlaindrome() receives a string with varying formats
+ * The function alphaNumericPalindrome() receives a string with varying formats
  * like "racecar", "RaceCar", and "race CAR"
  * The string can also have special characters
  * like "2A3*3a2", "2A3 3a2", and "2_A3*3#A2"
  *
  * But the catch is, we have to check only if the alphanumeric characters
  * are palindrome i.e remove spaces, symbols, punctuations etc
  * and the case of the characters doesn't matter
- *
- ****************************************************************************/
+ */
+const alphaNumericPalindrome = (str) => {
+  if (typeof str !== 'string') {
+    throw new TypeError('Argument should be string')
+  }
 
-const alphaNumericPlaindrome = (str) => {
   // removing all the special characters and turning everything to lowercase
-  const newStr = str.replace(/[^a-zA-Z0-9]*/g, '').toLowerCase()
+  const newStr = str.replace(/[^a-z0-9]+/ig, '').toLowerCase()
+  const midIndex = newStr.length >> 1
```

Comment:
if you want to keep it this way instead of a more readable `newStr.length / 2`, please add an explanatory comment (that this equals `floor(newStr.length / 2)`) even though this might seem somewhata self-explanatory to an advanced user of bitwise ops